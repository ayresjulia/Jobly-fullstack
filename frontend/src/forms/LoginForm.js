import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Alert from "../Alert";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";
import "./LoginForm.css";

/** Form to login a user and show homepage at "/" . */

const LoginForm = ({ login }) => {
	const history = useHistory();
	const [ formData, setFormData ] = useState({
		username: "",
		password: ""
	});
	const [ formErrors, setFormErrors ] = useState([]);

	async function handleSubmit (e) {
		e.preventDefault();
		let result = await login(formData);
		if (result.success) {
			history.push("/");
		} else {
			setFormErrors(result.errors);
		}
	}

	function handleChange (e) {
		const { name, value } = e.target;
		setFormData((formData) => ({ ...formData, [name]: value }));
	}

	return (
		<div className="LoginForm">
			<Form className="Form-body" onSubmit={handleSubmit}>
				<FormGroup>
					<Label htmlFor="username">Username</Label>
					<Input
						id="username"
						name="username"
						className="Form-input"
						value={formData.username}
						onChange={handleChange}
						autoComplete="username"
						required
					/>
				</FormGroup>

				<FormGroup>
					<Label htmlFor="password">Password</Label>
					<Input
						id="password"
						type="password"
						name="password"
						className="Form-input"
						value={formData.password}
						onChange={handleChange}
						autoComplete="current-password"
						required
					/>
				</FormGroup>

				{formErrors.length ? <Alert type="danger" messages={formErrors} /> : null}

				<Button className="btn btn-success float-right" onSubmit={handleSubmit}>
					Submit
				</Button>
			</Form>
		</div>
	);
};

export default LoginForm;
