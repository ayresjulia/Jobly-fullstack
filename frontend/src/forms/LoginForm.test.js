import { render } from "@testing-library/react";
import React from "react";
import LoginForm from "./LoginForm";

test("it renders without crashing", () => {
	render(<LoginForm />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<LoginForm />);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a field in the form", () => {
	const { getByText } = render(<LoginForm />);
	expect(getByText("Username")).toBeInTheDocument();
});
