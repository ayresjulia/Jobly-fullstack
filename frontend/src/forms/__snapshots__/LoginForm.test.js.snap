// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <form
      class="Form-body"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="username"
        >
          Username
        </label>
        <input
          autocomplete="username"
          class="Form-input form-control"
          id="username"
          name="username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="password"
        >
          Password
        </label>
        <input
          autocomplete="current-password"
          class="Form-input form-control"
          id="password"
          name="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="btn btn-success float-right btn btn-secondary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
